api/src/test/java/org/pmiops/workbench/access/UserServiceAccessTest.java

```diff
@@ -197,7 +203,61 @@ public void test_updateUserWithRetries_register_then_unregister() {
         .isEqualTo(TierAccessStatus.DISABLED);
   }
 
-  private DbUser registerUser(DbUser user) {
+  @Test
+  public void testUpdateUserWithRetriesForAccessRenewal() {
+    providedWorkbenchConfig.access.enableAccessRenewal = true;
+    providedWorkbenchConfig.access.enableDataUseAgreement = true;
+    providedWorkbenchConfig.accessRenewal.expiryDays = (long) 365;
+    DbUser dbUser = new DbUser();
+
+    // The user is still compliant and has bypass set
+    dbUser.setDataUseAgreementCompletionTime(
+        Timestamp.from(
+            START_INSTANT.minus(
+                (providedWorkbenchConfig.accessRenewal.expiryDays - 1), ChronoUnit.DAYS)));
+    dbUser.setDataUseAgreementSignedVersion(userService.getCurrentDuccVersion());
+    dbUser = userService.updateUserWithRetries(registerUserNow, dbUser, Agent.asUser(dbUser));
+
+    assertThat(userAccessTierDao.findAll()).hasSize(1);
+    assertThat(userAccessTierDao.getAllByUser(dbUser).get(0).getTierAccessStatusEnum())
+        .isEqualTo(TierAccessStatus.ENABLED);
+
+    // User is compliant and we remove dua bypass
+    dbUser.setDataUseAgreementCompletionTime(
+        Timestamp.from(
+            START_INSTANT.minus(
+                (providedWorkbenchConfig.accessRenewal.expiryDays - 1), ChronoUnit.DAYS)));
+    dbUser.setDataUseAgreementSignedVersion(userService.getCurrentDuccVersion());
+    dbUser =
+        userService.updateUserWithRetries(
+            this::registerUserWithoutDuaBypass, dbUser, Agent.asUser(dbUser));
+
+    assertThat(userAccessTierDao.findAll()).hasSize(1);
+    assertThat(userAccessTierDao.getAllByUser(dbUser).get(0).getTierAccessStatusEnum())
+        .isEqualTo(TierAccessStatus.ENABLED);
+
+    // The user DUA is no longer compliant
+    dbUser.setDataUseAgreementCompletionTime(
```

Comment:
nit: Moving the user's DUA completion time backwkards into the past isn't realistic, I would suggest any of the following alternatives:
1. Change the test case to instead cover going from expired -> unexpired (this is realistic, since it models the behavior of the user renewing their DUA)
2. If the goal is to test the switch from unexpired -> expired, I would instead keep the duaCompletionTime fixed, and simulate the passage of time via the fake clock.
3. If this is really meant to be 3 independent test cases (rather than tracking the state machine for one user over time), I'd split it into 3 test cases

Reply:
Updated to do #1 & #2. Walking through the flow of a user being compliant, time moving forward and they expire, followed up by a DUA update and they are compliant again.

Reply:
Nice, I like it.